src/core/portal/dom-portal-host.ts

```diff
@@ -0,0 +1,56 @@
+import {DynamicComponentLoader, AppViewManager, ComponentRef} from 'angular2/core';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {BasePortalHost, ComponentPortal, TemplatePortal} from './portal';
+
+
+/**
+ * A PortalHost for attaching portals to an arbitrary DOM element outside of the Angular
+ * application context.
+ *
+ * This is the only part of the portal core that directly touches the DOM.
+ */
+export class DomPortalHost extends BasePortalHost {
+  constructor(
+      private _hostDomElement: Element,
+      private _componentLoader: DynamicComponentLoader,
+      private _viewManager: AppViewManager) {
+    super();
+  }
+
+  /** Attach the given ComponentPortal to DOM element using the DynamicComponentLoader. */
+  attachComponentPortal(portal: ComponentPortal): Promise<ComponentRef> {
+    if (portal.origin == null) {
+      throw new BaseException(
+          'A ComponentPortal must have an origin set when attached to a DomPortalHost ' +
+          'because the DOM element is not part of the Angular application context.');
+    }
+
+    return this._componentLoader.loadNextToLocation(portal.component, portal.origin).then(ref => {
+      this._hostDomElement.appendChild(ref.hostView.rootNodes[0]);
+      this.setDisposeFn(() => ref.dispose());
+      return ref;
+    });
+  }
+
+  attachTemplatePortal(portal: TemplatePortal): Promise<Map<string, any>> {
+    let viewContainer = this._viewManager.getViewContainer(portal.templateRef.elementRef);
+    let viewRef = viewContainer.createEmbeddedView(portal.templateRef);
+    // TODO(jelbourn): locals don't currently work with DomPortalHost; investigate whether there
+    // is a bug in Angular.
```

Comment:
Have you tried to see if all locals actually get set properly? This is peculiar; looks like it should work.